File: app/src/main/java/tech/thdev/app/ui/photo/adapter/PhotoViewHolder.java
package tech.thdev.app.ui.photo.adapter;

import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import org.jetbrains.annotations.NotNull;

import butterknife.BindView;
import butterknife.ButterKnife;
import tech.thdev.app.R;
import tech.thdev.app.network.domain.Photo;
import tech.thdev.simpleadapter.holder.BaseViewHolder;

public class PhotoViewHolder extends BaseViewHolder<Photo> {

    @BindView(R.id.image)
    ImageView imageView;

    @BindView(R.id.text)
    TextView textView;

    public PhotoViewHolder(@NotNull ViewGroup parent, @NotNull OnItemClickListener onItemClickListener) {
        super(R.layout.item_photo_view, parent);

        ButterKnife.bind(this, itemView);

        itemView.setOnClickListener(v -> {
            onItemClickListener.onClickItem(getAdapterPosition());
        });
    }

    @Override
    public void onBindViewHolder(@NotNull Photo photo) {
        Glide.with(getContext())
                .load(String.format("https://farm%s.staticflickr.com/%s/%s_%s.jpg", String.valueOf(photo.farm), photo.server, photo.id, photo.secret))
                .centerCrop()
                .into(imageView);

        textView.setText(photo.title);
    }
}

File: kotlin-example-01/src/main/kotlin/tech/thdev/kotlin_example_01/view/adapter/view/PhotoViewHolder.kt
package tech.thdev.kotlin_example_01.view.adapter.view

import android.view.ViewGroup
import android.widget.ImageView
import com.bumptech.glide.Glide
import tech.thdev.kotlin_example_01.R
import tech.thdev.kotlin_example_01.base.adapter.BaseRecyclerAdapter
import tech.thdev.kotlin_example_01.base.adapter.BaseRecyclerViewHolder
import tech.thdev.kotlin_example_01.data.Photo

/**
 * Created by Tae-hwan on 7/22/16.
 */
class PhotoViewHolder(parent: ViewGroup?, adapter: BaseRecyclerAdapter<Photo>) : BaseRecyclerViewHolder<Photo>(R.layout.item_photo, parent, adapter) {

    private val imageView: ImageView

    init {
        imageView = itemView.findViewById(R.id.image_view) as ImageView
    }

    override fun onViewHolder(item: Photo?, position: Int) {
        Glide.with(context)
                .load(String.format("https://farm%s.staticflickr.com/%s/%s_%s.jpg", item?.farm, item?.server, item?.id, item?.secret))
                .placeholder(android.R.drawable.ic_menu_gallery)
                .centerCrop()
                .crossFade()
                .into(imageView)

        itemView.setOnLongClickListener {
            true
        }
    }
}